provision-osx/tasks/configure_buildfarm_node.yml

```diff
@@ -0,0 +1,139 @@
+---
+
+-
+  debug:
+    msg:
+    - Configuring macOS nodes as part of the build farm will be skipped.
+    - credential_private_key_path, credential_public_key_path and credential_passphrase must be defined.
+    - Run this job again to configure the build farm nodes.
+  when:
+  - credential_private_key_path is not defined or credential_public_key_path is not defined or credential_passphrase is not defined
+
+-
+  block:
+  -
+    name: Get jenkins url
+    shell: "oc get route jenkins --namespace={{ project_name }} -o jsonpath='{.spec.host}'"
+    delegate_to: "{{ groups['master'][0] }}"
+    register: route_output
+    changed_when: False
+
+  -
+    name: Get build farm env BASH
+    shell: "source ~/.bash_profile && echo $BASH"
+    register: buildfarm_env_bash
+    changed_when: False
+
+  -
+    name: Get build farm env GEM_HOME
+    shell: "source ~/.bash_profile && rvm gemdir"
+    register: buildfarm_env_gem
+    changed_when: False
+
+  -
+    name: Get build farm env NVM_PREFFIX
+    shell: "source ~/.bash_profile && echo $NVM_PREFFIX"
+    register: buildfarm_env_nvm
+    changed_when: False
+
+  -
+    name: Get build farm env PATH
+    shell: "source ~/.bash_profile && echo $PATH"
+    register: buildfarm_env_path
+    changed_when: False
+
+  -
+    block:
+    -
+      name: Get the jenkins client cli jar
+      connection: local
+      get_url:
+        url: "{{ jenkins_route_protocol | default('https') }}://{{ route_output.stdout }}/jnlpJars/jenkins-cli.jar"
+        dest: /tmp/jenkins-cli.jar
+        force: yes
+        validate_certs: no
+        mode: 0775
+      environment:
+        http_proxy: "{{ proxy_url | default('') }}"
+        https_proxy: "{{ proxy_url | default('') }}"
+
+    -
+      name: Retrieve provided private key
+      shell: "cat {{ credential_private_key_path }}"
+      register: private_key
+      changed_when: False
+
+    -
+      name: Retrieve provided public key
+      shell: "cat {{ credential_public_key_path }}"
+      register: public_key
+      changed_when: False
+
+    -
+      name: Set credential_private_key and credential_passphrase
+      set_fact:
+        credential_private_key: "{{ private_key.stdout }}"
+        credential_public_key: "{{ public_key.stdout }}"
+        credentials_provided: true
```

Comment:
Not sure why we need to do these three tasks? Looks like your just reassigning a variable name that we already have a value for?